File: padroesCriacionais/padrao-de-projeto-factory/java/src/test/java/MercadoPagoTest.java
import factory.MercadoPago;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

public class MercadoPagoTest {

    public MercadoPago mercadoPago;

    @BeforeEach
    public void setUp() {
        this.mercadoPago = new MercadoPago();
    }

    @Test
    public void deveRealizarPedidoPixComSucesso() {
        String response = this.mercadoPago.pix("dados do pedido", "dados do usuario");
        assertEquals("dados do pagamento com pix no Mercado Pago", response);
    }

    @Test
    public void deveRealizarPedidoBoletoComSucesso() {
        String response = this.mercadoPago.boleto("dados do pedido", "dados do usuario");
        assertEquals("dados do pagamento com boleto no Mercado Pago", response);
    }

    @Test
    public void deveRealizarPedidoCreditoComSucesso() {
        String response = this.mercadoPago.credito("dados do pedido", "dados do usuario");
        assertEquals("dados do pagamento com credito no Mercado Pago", response);
    }
}

File: padroesCriacionais/padrao-de-projeto-factory/php/tests/unit/MercadoPagoTest.php
<?php

namespace Factory\Test;

use Factory\Php\MercadoPago;
use PHPUnit\Framework\TestCase;

class MercadoPagoTest extends TestCase
{
    private ?MercadoPago $mercadoPago = null;

    /**
     * @before
     */
    public function setUp(): void {
        $this->mercadoPago = new MercadoPago();
    }

    public function testDeveRealizarPedidoPixComSucesso()
    {
        $response = $this->mercadoPago->pix("dados do pedido", "dados do usuario");
        $this->assertEquals("dados do pagamento com pix no Mercado Pago", $response);
    }

    public function testDeveRealizarPedidoBoletoComSucesso()
    {
        $response = $this->mercadoPago->boleto("dados do pedido", "dados do usuario");
        $this->assertEquals("dados do pagamento com boleto no Mercado Pago", $response);
    }

    public function testDeveRealizarPedidoCreditoComSucesso()
    {
        $response = $this->mercadoPago->credito("dados do pedido", "dados do usuario");
        $this->assertEquals("dados do pagamento com credito no Mercado Pago", $response);
    }
}